spec/lib/ansible/runner_execution_spec.rb

```diff
@@ -8,10 +8,17 @@
 
   describe ".run" do
     it "runs a hello-world playbook" do
-      response = Ansible::Runner.run(env_vars, extra_vars, File.join(__dir__, "runner/data/hello_world.yml"))
+      response = Ansible::Runner.run(env_vars, extra_vars, File.join(__dir__, "runner/data/hello_world.yml"), test_vault: false)
 
       expect(response.return_code).to eq(0), "ansible-runner failed with:\n#{response.stderr}"
       expect(response.human_stdout).to include('"msg": "Hello, world!"')
     end
+
+    it "runs a hello-world-vault-encrypted playbook" do
+      response = Ansible::Runner.run(env_vars, extra_vars, File.join(__dir__, "runner/data/hello_world_vault_encrypted.yml"), test_vault: true)
```

Comment:
You should be able to create a VaultCredential directly, and there is no need for the test_vault parameter.

```suggestion
      credential = FactoryBot.create(:embedded_ansible_vault_credential, :password => "vault")
      playbook = File.join(__dir__, "runner/data/hello_world_vault_encrypted.yml")
      response = Ansible::Runner.run(env_vars, extra_vars, playbook, :credentials => [credential.id])
```